contracts/src/interfaces/IPayer.sol

```diff
@@ -264,54 +261,37 @@ interface IPayer {
     //==============================================================
 
     /**
-     * @notice Called by node operators to settle usage and calculate fees owed.
-     * @dev This function is EIP-2200 optimized by using accumulators for multiple state updates.
-     * @param fees The total USDC fees computed from this usage period.
-     * @param payer The address of the payer being charged.
-     * @param nodeId The ID of the node operator submitting the usage.
-     * @param timestamp The timestamp when the usage occurred (can be backdated).
-     *
-     * Emits `UsageSettled`.
-     */
+    * @notice Settles usage for a contiguous batch of (payer, amount) entries.
+    * Assumes that the PayerReport contract has already verified the aggregated Merkle proof.
+    *
+    * @param originatorNode The node that submitted the report.
+    * @param reportIndex The index of the report.
+    * @param offset The starting index of the batch in the report's data.
+    * @param payers A contiguous array of payer addresses.
+    * @param amounts A contiguous array of usage amounts corresponding to each payer.
+    */
     function settleUsage(
-        uint256 fees,
-        address payer,
-        uint256 nodeId,
-        uint256 timestamp
-    ) external;
-
-    /**
-     * @notice Called by node operators to settle usage for multiple payers in a batch.
-     * @dev Uses EIP-2200 optimizations for storage efficiency.
-     * @param payers Array of payer addresses being charged.
-     * @param fees Array of USDC fees corresponding to each payer.
-     * @param timestamp When this batch of usage occurred (can be backdated).
-     * @param nodeId The ID of the node operator submitting the usage.
-     *
-     * Emits `BatchUsageSettled` and multiple `UsageSettled` events.
-     */
-    function settleUsageBatch(
+        address originatorNode,
```

Comment:
I believe this will also need to include a Batch Merkle Proof, won't it?

Reply:
Actually I think offset is not needed in Payer.settleUsage function.
The way I see it is the PayerReport.settleUsageBatch accepts the proofs and verifies them. Only when the []payers and []amounts are verified as part of the merkle tree we call Payer.settleUsage([]payer, []amount), which is a dumb function just to settle what we agreed on.